### ISSUE TITLE
Can only upload text files.

### ISSUE THREAD

**Issue by narayanjr** [Kaliber/play-s3#70]
Using Play 2.4.1 and play-s3 version 7.0.2

I am able to upload text files just fine, but when I try anything more complex, images, .ods, .jar files it fails with this error `The provided 'x-amz-content-sha256' header does not match what was computed.`

If I swap "avatar.jpeg" for "text.txt" and "image/jpeg" for "plain/text" in the following code it works. The text file gets uploaded to S3. But if I try and upload any thing else it fails saying the header doesn't match. I've tried png, jpg, jpeg, ods,  and jar files.

    val file_path = "/path/to/file/avatar.jpeg"
    val bucket = S3("path_to_bucket")
    val byte_array = Files.readAllBytes(Paths.get(file_path))

    val result = bucket + BucketFile("avatar.jpeg", "image/jpeg", byte_array)
    result.map { unit =>
        Logger.info("Saved the file")
    }
        .recover {
            case S3Exception(status, code, message, originalXml) =>
            {
                Logger.info("Error: " + message)
                Logger.info("originalXml: " + originalXml)
            }
        }

**Comment by EECOLOR** [comment]
@narayanjr Let me see if I can reproduce the problem.

**Comment by EECOLOR** [comment]
I can not reproduce the problem and I can upload my slack avatar without a problem. Could you post the result from your log file?

**Comment by narayanjr** [comment]
This is the "original xml" logged message and the example image that generated this message.
![avatar](https://cloud.githubusercontent.com/assets/6971369/18595044/edb350d0-7c10-11e6-8b4f-9325e0230614.jpeg)

    <Error>
       <Code>XAmzContentSHA256Mismatch</Code>
       <Message>The provided 'x-amz-content-sha256' header does not match what was computed.</Message>
       <ClientComputedContentSHA256>6d32a470a69d0c3af4b345e8a543ede0a425180bca616eba903ba32f35ee31b2</ClientComputedContentSHA256>
       <S3ComputedContentSHA256>949d177392b7651e8c1fa67239a777f2b9fb2e6f5b9898e83a74dab01f5f010c</S3ComputedContentSHA256>
       <RequestId>16ACAA3A2A2AE950</RequestId>
       <HostId>QJUy+PYrgOuFKw6mIwDBA9WVXIkRJH3cQx/znKhqCT0X9LtwVmc6bmmH3HJqO/38znbGvXtp0TA=</HostId>
    </Error>

**Comment by narayanjr** [comment]
Hmmm, I just created another play project just to test this single thing and it seems to work as expected there.  So there is something wrong with my other app.  However I have no idea how/why it's sha256 hash could be different.

**Comment by EECOLOR** [comment]
Ohw, that is nasty. Those types of bugs are very hard to trace.

A few things to check:
- Check if the security providers are the same (https://docs.oracle.com/javase/8/docs/api/java/security/Security.html#getProviders--)
- Check if the `Array[Byte]` is the same before and after the `Future` completes
- Check your configuration for a proxy that might change the request
- Check if you have the problem in the other play application with the same file (it might contain some weird bytes)
- Check for differences in `play.ws` properties in the configuration

I can not think of anything else. If none of the above works, remove halves of the code until you find the offending part (binary search). Please report back once you found the problem so others don't need to go through the same struggle.

**Comment by narayanjr** [comment]
So I ended up resorting to cloning the broken project and slowly working my way to making both the working example and broken project look identical.  After doing this for a few hours I finally managed to find the 1 thing that was different between the 2 projects.

My working example was using Play 2.4.2 while my broken App was using 2.4.1.

Here is a bare minimum example of this issue, https://github.com/narayanjr/play-s3-test

I have no idea why the issue exists but changing the Play version fixes it.
